src/products/domain/stock.domain.ts

```diff
@@ -0,0 +1,32 @@
+import {
+  Column,
+  Entity,
+  JoinColumn,
+  ManyToOne,
+  PrimaryGeneratedColumn,
+} from 'typeorm';
+import { Product } from './product.domain';
+import { Location } from './location.domain';
+
+@Entity()
+export class Stock {
+  @PrimaryGeneratedColumn('uuid', { name: 'LocationId' })
+  locationId: string;
+
+  @PrimaryGeneratedColumn('uuid', { name: 'ProductId' })
+  productsId: string[];
```

Comment:
The `Stock` table is used to create a many-to-many relation between `Location` and `Product`. 

One stock item should correspond to a single location and a single product, so not a list of products (similar to what you did for `OrderDetail` table). Both `locationId` and `productId` should be strings, not arrays of strings.

Besides that, `locationId` and `productId` should not be autogenerated, they should be provided because they should match some items from `Location` and `Product` table.